### ISSUE TITLE
[Help Wanted] [PLT-7728] Create Jenkinsfile for mattermost-desktop and create build on new build server

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost-server#7517]
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://docs.mattermost.com/guides/developer.html), specifically for [machine setup](https://docs.mattermost.com/developer/developer-setup.html) and for [developer workflow](https://docs.mattermost.com/developer/developer-flow.html).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/PLT-7728)

Below is the current build servers main build script. 

```
#!/bin/bash
set -e

cd src/github.com/mattermost/desktop

npm install
npm run build

rm -rf release

# Build and compress for Linux
npm run package:linux
mv release/mattermost-desktop-*-ia32.tar.gz  release/tmp-linux-ia32.tar.gz
mv release/mattermost-desktop-*.tar.gz release/mattermost-desktop-linux-x64.tar.gz
mv release/tmp-linux-ia32.tar.gz release/mattermost-desktop-linux-ia32.tar.gz
mv release/mattermost-desktop_*_i386.deb release/mattermost-desktop-linux-i386.deb
mv release/mattermost-desktop_*_amd64.deb  release/mattermost-desktop-linux-amd64.deb

# Build and compress for OSX
npm run package:mac
mv release/mac/Mattermost-*-mac.tar.gz release/mattermost-desktop-osx.tar.gz

# Build and compress for Windows
npm run package:windows
mv release/Mattermost-*-ia32-win.zip release/mattermost-desktop-win32.zip
mv release/Mattermost-*-win.zip release/mattermost-desktop-win64.zip
mv release/win-ia32/Mattermost\ Setup\ *-ia32.exe release/mattermost-setup-win32.exe
mv release/win/Mattermost\ Setup\ *.exe release/mattermost-setup-win64.exe
```

**Comment by cpanato** [comment]
we will move the build to jenkins or will keep it running on circleci?

**Comment by jasonblais** [comment]
Posted in Desktop App channel @cpanato https://pre-release.mattermost.com/core/pl/cu1oeccggbn7zmxtde4gha1yra

**Comment by Brunzer** [comment]
I'll grab this is if it still available

**Comment by jasonblais** [comment]
Thanks @Brunzer! @MusikPolice Is this relevant to what you've been working on past week, or is this different?

**Comment by jasonblais** [comment]
Just don't want to have you and @Brunzer working on the same thing in parallel.

**Comment by Brunzer** [comment]
It looks like there's already a Jenkinsfile in [desktop](https://github.com/mattermost/desktop) but I didn't see a job created yet on [mattermost-build](https://github.com/mattermost/mattermost-build). Seems like this might already be mostly completed :smiley:

**Comment by MusikPolice** [comment]
Yes, I've migrated the [circle.yml](https://github.com/mattermost/desktop/blob/master/circle.yml) file to a [Jenkinsfile](https://github.com/mattermost/desktop/blob/PLT-7540/build/Jenkinsfile) as a part of [PLT-7540](https://mattermost.atlassian.net/browse/PLT-7540). I'm currently working on adding build signing to the script.

**Comment by jasonblais** [comment]
@MusikPolice Should we then close this help wanted ticket and assign the ticket to you, or close as a duplicate of PLT-7540?

**Comment by MusikPolice** [comment]
Closing as a dupe.
